custom_components/ltss/__init__.py

```diff
@@ -267,46 +267,77 @@ def event_listener(self, event):
             if self.entity_filter(entity_id):
                 self.queue.put(event)
 
+    def _ltss_is_hyptertable(self):
+        if 'timescaledb' not in self.available_extensions:
+            return False
+
+        return 1 == self.con.execute(
+            text(
+                f"SELECT 1 FROM timescaledb_information.hypertables "
+                f"WHERE hypertable_name = '{LTSS.__tablename__}'")).rowcount
+
+    def _ltss_with_postgis(self):
+        return 1 == self.con.execute(
+            text(
+                f"SELECT 1 FROM information_schema.columns "
+                f"WHERE table_name = '{LTSS.__tablename__}' AND COLUMN_NAME = 'location'")).rowcount
+
+    def _update_chunk_time_interval(self):
+        self.con.execute(text(f"SELECT set_chunk_time_interval('{LTSS.__tablename__}', {self.chunk_time_interval});")
+                         .execution_options(autocommit=True)
+                         )
+
```

Comment:
In my opinion, these functions does not really help the understanding of the `setup_connection`. I would rather see that we leverage the existing inspection capabilities of `sqlalchemy`, i.e. something along these lines (put directly within the `setup_connection` method):
```python
inspector = inspect(self.engine)
if inspector.has_table(LTSS.__table__name):
    columns = [column_conf["name"] for column_conf in inspector.get_columns(LTSS.__tablename__)]
    if "location" in columns:
        # do something

```